packages/webpack-cli/lib/utils/cli-flags.js

```diff
@@ -286,11 +287,16 @@ module.exports = {
         {
             name: 'mode',
             usage: '--mode <development | production>',
-            type: String,
+            type: (value) => {
+                if (value === 'development' || value === 'production' || value === 'none') {
+                    return value
+                } else {
+                    logger.warn('You provided invalid value for "mode" option, webpack will fallback to "production" for this value.')
```

Comment:
```suggestion
                    logger.warn('You provided an invalid value for "mode" option.')
```

Reply:
Let's not fallback to "production". Let's not pass anything, so also `webpack` will provide a meaningful warning on how to fix it.